lib/ruby_terraform/command_line/option.rb

```diff
@@ -0,0 +1,60 @@
+require 'json'
+
+module RubyTerraform
+  module CommandLine
+    class Option
+      def initialize(opt_key, value, switch_override: nil)
+        assign_option_value(value)
+        @switch = switch_override || switch_from_key(opt_key)
+      end
+
+      def add_to_subcommand(sub)
+        if value.respond_to?(:keys)
+          add_hash_to_subcommand(sub, value)
+        elsif value.respond_to?(:each)
+          add_array_to_subcommand(sub, value)
+        else
+          add_option_to_subcommand(sub)
+        end || sub
+      end
+
+      private
+
+      attr_reader :switch, :value
+
+      def assign_option_value(value)
+        @value = value
+      end
+
+      def switch_from_key(opt_key)
+        "-#{opt_key.to_s.sub('_', '-')}"
+      end
+
+      def add_hash_to_subcommand(sub, hash)
+        hash.each do |hash_key, hash_value|
+          sub = sub.with_option(switch, "'#{hash_key}=#{as_string(hash_value)}'", separator: ' ')
```

Comment:
Hard to see how this could be pushed down to `lino` since the format of the values is specific-ish to terraform. I wonder though if `lino` should have a `#with_options` or `#with_repeated_option` that takes an array of values and adds them all for the same switch. That way, this class would do the transformation of the values and `lino` would handle constructing the command line.